File: problems/1911_maximum_alternating_subsequence_sum/solution.py
from typing import List


class Solution:
    def maxAlternatingSum(self, nums: List[int]) -> int:
        even = 0
        odd = 0
        for x in nums:
            new_even = max(even, odd + x)
            new_odd = max(odd, even - x)
            even = new_even
            odd = new_odd
        return even

File: problems/1911_maximum_alternating_subsequence_sum/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    long long maxAlternatingSum(vector<int>& nums) {
        long long even = 0, odd = 0;
        for (int x : nums) {
            long long newEven = max(even, odd + x);
            long long newOdd = max(odd, even - x);
            even = newEven;
            odd = newOdd;
        }
        return even;
    }
};
